action.yml

```diff
@@ -31,5 +31,45 @@ runs:
         allow-prereleases: true
 
     - name: "Install nox"
-      run: pipx install --python "${{ steps.allpython.outputs.python-path }}" '${{ github.action_path }}'
-      shell: bash
+      run: |
+        import os
+        import shutil
+        import sys
+        import venv
+
+        from pathlib import Path
+        from subprocess import run
+
+
+        class EnvBuilder(venv.EnvBuilder):
+            def __init__(self):
+                super().__init__()
+
+            def setup_scripts(self, context):
+                pass
+
+            def post_setup(self, context):
+                super().post_setup(context)
+                self.bin_path = Path(context.env_exe).parent
+                run([sys.executable, "-m", "pip", "--python", context.env_exe, "install", r"${{ github.action_path }}"], check=True)
+
+
+        print("::group::Install nox")
+        nox_bin_path = Path(r"${{ runner.temp }}") / "nox"
+        if nox_bin_path.exists():
+            shutil.rmtree(nox_bin_path)
+        venv_path = Path(r"${{ runner.temp }}") / "nox-venv"
+        if venv_path.exists():
+            shutil.rmtree(venv_path)
+        builder = EnvBuilder()
+        builder.create(venv_path)
+        nox_path = [path for path in builder.bin_path.glob("nox*") if path.stem == "nox"][0]
+        nox_bin_path.mkdir()
+        try:
+            os.symlink(nox_path, nox_bin_path / nox_path.name)
+        except OSError:
+            shutil.copyfile(nox_path, nox_bin_path / nox_path.name)
+        with open(os.environ["GITHUB_PATH"], "at") as f:
+            f.write(f"{nox_bin_path}\n")
+        print("::endgroup::")
+      shell: "${{ steps.localpython.outputs.python-path }}" -u
```

Comment:
```suggestion
      shell: "${{ steps.localpython.outputs.python-path }} -u"
```

Reply:
I used the shell trick because you can't quote the python path (`shell: "'${{ steps.localpython.outputs.python-path }}' -u"`). Maybe I was being too paranoid over self-hosted runners getting spaces in their path and that should not really matter.

Reply:
Oh, okay. Worth reverting? Why can’t it be quoted, also?

Reply:
It's not worth reverting. I'll try to link/open a relevant issue in the runner implementation. I guess they just split by space but do not use any shlex.split resulting in the quoted string being used as the real path and fail to run the executable.